pkg/lib/cockpit.d.ts

```diff
@@ -98,6 +99,8 @@ declare module 'cockpit' {
         changed(): void;
     }
 
+    function event_target(obj: unknown): unknown;
```

Comment:
This feels a little too vague -- this should capture the important bit that it spits out the same type as you put in. So perhaps
```js
function event_target<T>(obj: T): T;
```
? (Note, I'm a ts novice as well).

That's still not quite right of course -- the correct thing to return would be some kind of "`T` with benefits", like an extra interface that has `.addEventListener()` and such.

Reply:
Yes, I was thinking the same... we would need to use it like `return cockpit.event_target(self)` then. We don't do that anywhere, I think, but it would be an improvement. I will try it out.

On the other hand, I start to think that we need to convert all code to use `class Foo extends EventTarget` instead. I would expect TypeScript to be way better with explicit classes than with the closures-as-classes pattern...

Reply:
Yes, right. This seems to quickly get into yak shaving, I was just surprised that an unknown → unknown annotation is actually good for anything? Or is that just a way to say "shut up tsc"?

Reply:
> I was just surprised that an unknown → unknown annotation is actually good for anything? Or is that just a way to say "shut up tsc"?

It's not good for much, but it let's you call this function. It doesn't shut up Tsc: It wont let you do anything with the return value (without casting it as something else), and it will check the number of arguments at least. But it will not make TypeScript aware that the object passed into event_target now has a addEventListener function, etc.

So, yeah, it's not good for much. We need to switch to classes and inherit from EventTarget.

Reply:
> We need to switch to classes and inherit from EventTarget.

Or better yet, from our own EventEmitter. I'll make a `useOn` hook for that...